build_tools/amdgpu_family_matrix.py

```diff
@@ -0,0 +1,33 @@
+amdgpu_family_info_matrix = {
```

Comment:
Without context this Python "script" is a little confusing as it is not obvious what it is intended for. A short comment would help. We should also discuss how we want to organize the files. In IREE, we have `build_tools/github_actions` for example.